testsuite/integration/multinode/src/test/java/org/jboss/as/test/multinode/transaction/async/ClientBean.java

```diff
@@ -84,10 +84,28 @@ public void callToStatusByTransactionmanager() throws Exception {
         final TransactionalRemote remote = getRemote(TransactionalStatusByManager.class);
         userTransaction.begin();
         try {
-            Assert.assertEquals("No transaction expected as async call does not pass txn context",
+            assertEquals("No transaction expected as async call does not pass txn context",
                     (Integer) Status.STATUS_NO_TRANSACTION, remote.transactionStatus().get());
         } finally {
             userTransaction.rollback();
         }
     }
+
+    /**
+     * Verifies async method invocation with REQUIRED transaction attribute.
+     * The client transaction context should not be propagated to the invoked async method.
+     * The invoked async method should execute in a new, separate transaction context.
+     */
+    public void callToRequired() throws Exception {
+        final TransactionalRemote remote = getRemote(TransactionalStatusByManager.class);
+        userTransaction.begin();
+
+        // asyncWithRequired() will throw RuntimeException and cause its transaction to rollback.
+        // But it has no bearing on the transaction here, which should be able to commit okay.
+        remote.asyncWithRequired();
+
+        // wait a bit for the transaction in asyncWithRequired method to rollback
+        Thread.sleep(1000);
```

Comment:
You hold the Future, so better use it.

Reply:
`asyncWithRequired` method returns void since it just throws a `RuntimeException` with no need to return a value. We could add a return value and use `get()` instead of `sleep()`.

Reply:
Get instead of sleep would definitely be preferred as sleep could be racy.

Reply:
If it did return void, then we have another issue. It should be throwing an `ExecutionException`.
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/ExecutionException.html

Reply:
When it returns void, I think it should just be a fire-and-forget on the client side, and whatever happened in the target method is not visible to the client at all.  The `ExecutionException` is available via the `get()` method when the method returns a `Future`.